app.js

```diff
@@ -10,6 +11,16 @@ app.get("/api/topics", getAllTopics);
 
 app.get("/api", getEndpoints)
 
+app.get("/api/articles/:article_id", getArticleById)
+
+app.use((err, req, res, next) => {
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg });
+  }
+  
+  res.status(404).send();
```

Comment:
this second res should be within an else block as currently this may try to run after the if block is executed, and you may face an error saying 'cannot set headers after they are sent'.
Also rather than sending a 404 here we should invoke next() with the error and send a 500, internal server error in the next app.use().
Before this current app.use() we should also have one that handle PSQL error. Please refer back to the NC Notes on error handling middleware